code/__DEFINES/misc.dm

```diff
@@ -253,7 +253,8 @@
 #define MATRIX_DEFAULT list(1, 0, 0, 0,\
                             0, 1, 0, 0,\
                             0, 0, 1, 0,\
-                            0, 0, 0, 1)
+                            0, 0, 0, 1,\
+                            0, 0, 0, 0)
```

Comment:
Without having the matrices the lengths they are, the logic that determines whether a person's client.colour == or != whatever the colour matrix that's being applied is doesn't work.

Good point, going to gut this.